### ISSUE TITLE
Island Chest reset

### ISSUE THREAD

**Issue by WarnDa** [BG-Software-LLC/SuperiorSkyblock2#622]
### Minecraft's Version

1.16.5

### Plugin's Version

b316

### Describe the bug

Sometimes, island chests are reset for some players.

In 48h, islands chest were clear 4 times

### To Reproduce

Put items in your island chest, and wait (maybe after a restart ?)

### Additional Information

Using MySQL

**Comment by OmerBenGera** [comment]
Have you been able to replicate the issue yourself?
I don't see any instructions on how to replicate it.

**Comment by WarnDa** [comment]
> Have you been able to replicate the issue yourself?

Yeah, yesterday I had a block in my /is chest
and tomorrow there is no block anymore.

So maybe it's during the restart, I can't try again currently as I won't restart with + 50 players.

But I'll re-put a block and see tomorrow

**Comment by OmerBenGera** [comment]
> 
> 
> > Have you been able to replicate the issue yourself?
> 
> Yeah, yesterday I had a block in my /is chest
> and tomorrow there is no block anymore.
> 
> So maybe it's during the restart, I can't try again currently as I won't restart with + 50 players.
> 
> But I'll re-put a block and see tomorrow

Try it on another server and not on your productive server.
Let me know when it happens.

**Comment by WarnDa** [comment]
I tried to reproduce it but I can't.

But might be related to https://github.com/BG-Software-LLC/SuperiorSkyblock2/issues/619

**Comment by OmerBenGera** [comment]
The other issue have chests disappear, but chests with items are saved.
I am not sure if it's related, therefore we'll wait for a fix to be published for the other issue and we'll see if this issue fixed as well.

**Comment by OmerBenGera** [comment]
Does it happen with every restart?

**Comment by WarnDa** [comment]
> Does it happen with every restart?

Yes, every time I restart, I tried on multiple account, everyrone is reset

**Comment by OmerBenGera** [comment]
Mind testing it using SQLite on another server?

**Comment by WarnDa** [comment]
I can't reproduce the issue in another server :/

**Comment by OmerBenGera** [comment]
> 
> 
> I can't reproduce the issue in another server :/

It only happens on your server?

**Comment by OmerBenGera** [comment]
When you create an island and put items in your chest, when the server stops - do you have any island chests in the database?

**Comment by OmerBenGera** [comment]
note tho, when putting items in island chests, they won't be saved instantly.

**Comment by WarnDa** [comment]
My islands_chests database is just like this :
https://cdn.discordapp.com/attachments/411621057892581376/893179445697982484/unknown.png

nothing else

**Comment by WarnDa** [comment]
(And on my test server it works without issues : )
https://cdn.discordapp.com/attachments/411621057892581376/893179756755955822/unknown.png

**Comment by OmerBenGera** [comment]
> 
> 
> My islands_chests database is just like this : https://cdn.discordapp.com/attachments/411621057892581376/893179445697982484/unknown.png nothing else
> 
> But I added items 5 hours ago in my chest : ![Screenshot_6486](https://user-images.githubusercontent.com/31826333/135498694-164d4d5d-8e0f-42d7-97ea-8aa92a7f230d.png)
> 
> So it doesn't save anything related to islands_chests. I'll try to delete database

How do you know if anything was saved from this picture?
Use `Action: Database Execute, Query: UPDATE islands_chests` and check if anything is printed to screen.
Should be printed every now or then. Anyways, chests are saved when server is stopped.

**Comment by WarnDa** [comment]
> How do you know if anything was saved from this picture?

Because even 2 hours after, database is still clear

**Comment by WarnDa** [comment]
> Use Action: Database Execute, Query: UPDATE islands_chests and check if anything is printed to screen.
Should be printed every now or then. Anyways, chests are saved when server is stopped.

I'll test  tomorrow, not at home this night

**Comment by OmerBenGera** [comment]
> > How do you know if anything was saved from this picture?
> 
> Because even 2 hours after, database is still clear

Database is cleared..?

**Comment by OmerBenGera** [comment]
Any news regarding this?

**Comment by maikijs** [comment]
https://pastebin.com/YETX0K5U

**Comment by maikijs** [comment]
actually, i lied. After testing the issue with block limit not working (#587), found out that here is the same issue. If you enable upgrades in (\modules\upgrades) then is chest is saving properly. 👍

**Comment by WarnDa** [comment]
Ohh, I got upgrades disabled, so that might explain the issue !
